torch/testing/_comparison.py

```diff
@@ -333,25 +333,36 @@ def __init__(
         self.id = id
         self._unknown_parameters = unknown_parameters
 
+    @staticmethod
+    def _inputs_not_supported() -> NoReturn:
+        raise UnsupportedInputs()
```

Comment:
Although, this looks quite unnecessary, it has one major advantage: raising `UnsupportedInputs` directly, leaks implementation details. For example, we no longer need to import `UnsupportedInputs` for `TestCase.assertEqual` and in there no one has to know what happens internally.